### PR TITLE
feat: Backend for new models contract

### LINKED ISSUE #1050: [Task]: Model Contract - Application Backend — Delete Handlers, Simplify Factory
### Description

With the new backend-agnostic `Model` contract (numpy in/out, no handler indirection), the application backend no longer needs `TorchModelHandler` / `OpenVINOModelHandler`. The factory returns `Model` directly, and the processor calls `model.predict(batch)` without backend-specific branching.

This issue can be tackled only after the library side things are dine.

#### 1. Delete model handlers

| File | Action |
|------|--------|
| `application/backend/app/runtime/core/components/models/torch_model.py` | Delete |
| `application/backend/app/runtime/core/components/models/openvino_model.py` | Delete |
| `application/backend/app/runtime/core/components/models/passthrough_model.py` | Keep (or simplify) |

These handlers currently:
- `TorchModelHandler`: wraps library model, converts numpy→tensor for input, tensor→numpy for output, calls `model.predict(batch)`
- `OpenVINOModelHandler`: exports model to OV, compiles, resizes masks back to frame size, derives bboxes from masks

With the new contract, the model itself handles all of this internally. No handler needed.

#### 2. Simplify factory (`application/backend/app/runtime/core/components/factories/model.py`)

**Before:** Factory creates library model + wraps in handler based on `processor_openvino_enabled` setting.

**After:**
```python
def create_model(config: ModelConfig) -> Model:
    """Factory returns Model directly. No handler wrapping."""
    match config:
        case MatcherConfig():
            model = Matcher(...)
        case Sam3Config() if config.use_openvino:
            model = SAM3OpenVINO(...)
        case Sam3Config():
            model = SAM3(...)
        ...
    return model
```

No more `TorchModelHandler(model)` or `OpenVINOModelHandler(model)` wrapping.

#### 3. Update processor (`application/backend/app/runtime/core/components/processor.py`)

**Before:**
```python
results = self._model_handler.predict(batch_data)  # returns list[dict[str, np.ndarray]]
```

**After:**
```python
samples = [Sample(image=frame.image) for frame in batch_data]
predictions = self._model.predict(samples)  # returns list[Prediction]
```

- Remove `_model_handler` — replace with `_model: Model`
- Remove `InputData` → numpy conversion (model handles it)
- Remove mask resizing logic (model's postprocessor handles it)
- Map `Prediction` → `OutputData` for downstream consumers

#### 4. Remove preprocessing from backend

The backend currently does frame resizing before passing to handlers. With the new contract, models own their own preprocessor pipeline — backend just passes raw frames.

#### 5. Update base interface (`application/backend/app/runtime/core/components/base.py`)

- Remove or simplify `ModelHandler` ABC
- The processor now holds a `Model` instance directly

#### 6. Update imports

- `from instantlearn.models import Model, SAM3, Matcher, ...`
- `from instantlearn.data import Sample, Prediction`
- Remove: `from instantlearn.data.base.batch import Batch`

### Files to modify

| File | Action |
|------|--------|
| `app/runtime/core/components/models/torch_model.py` | Delete |
| `app/runtime/core/components/models/openvino_model.py` | Delete |
| `app/runtime/core/components/factories/model.py` | Simplify |
| `app/runtime/core/components/processor.py` | Update predict call |
| `app/runtime/core/components/base.py` | Remove/simplify ModelHandler ABC |
| `app/domain/services/` | Update if referencing old types |
| Tests in `application/backend/tests/` | Update for new return types |

### Component

Library (instantlearn Framework)

### Acceptance Criteria

- [ ] `TorchModelHandler` and `OpenVINOModelHandler` deleted
- [ ] Factory returns `Model` directly (no handler wrapping)
- [ ] Processor calls `model.predict(list[Sample])` → gets `list[Prediction]`
- [ ] No numpy→tensor→numpy round-trips in backend
- [ ] No mask resizing in backend (model's postprocessor handles it)
- [ ] Backend tests pass
- [ ] End-to-end inference works (WebRTC stream → predictions → output)

### Implementation Notes

_No response_

### Parent Epic

_No response_

### Dependencies

_No response_

### PR DESCRIPTION
**Pull request by LenaRudaya**
This PR updates the backend runtime to a new “models contract” centered on `instantlearn` `Prediction` objects (instead of legacy `pred_* dicts`), refactors model construction/export (including `SAM3OpenVINO` IR caching), and hardens pipeline lifecycle/concurrency behavior in `PipelineManager`.

Changes:

- Replaced legacy dict-based model outputs with Prediction across pipeline, visualizer overlays, MQTT serialization, and tests.
- Introduced a unified InferenceModelHandler and refactor ModelFactory to route Torch vs OpenVINO by device type, including an on-disk SAM3 IR cache.
- Reworked PipelineManager locking/state to keep status polling/WebRTC access responsive during long model builds, with build cancellation and more robust lifecycle semantics.

## Type of Change

- [x] ✨ `feat` - New feature
- [ ] 🐞 `fix` - Bug fix
- [ ] 📚 `docs` - Documentation
- [x] ♻️ `refactor` - Code refactoring
- [x] 🧪 `test` - Tests
- [ ] 🔧 `chore` - Maintenance

## Related Issues
Closes #1050

### CODE REVIEW

File: library/tests/unit/models/test_sam3_exemplar.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def test_default_classic_mode(self, _mock_tokenizer: MagicMock, _mock_model: Mag
         """SAM3 defaults to CLASSIC prompt mode."""
         sam3 = SAM3(device="cpu")
         assert sam3.prompt_mode == Sam3PromptMode.CLASSIC
-        assert sam3.drop_spatial_bias is False
+        assert sam3.drop_spatial_bias is True
```
Comment (michalad1) [inline library/tests/unit/models/test_sam3_exemplar.py]: I suppose this is from bad merge?
because by default drop_spatial_bias is set to False in SAM3.

File: application/backend/app/runtime/core/components/factories/model.py [hunk 2]
```diff
@@ -1,145 +1,455 @@
...
+        reference_batch: Batch,
+        device_info: DeviceInfo,
+        *,
+        use_openvino: bool,
+    ) -> Model | None:
+        """Construct and fit a model, converting it to OpenVINO when enabled."""
+        selected_device = device_info.as_torch
+        # If the model is converted to the OV format the precision must be fp32:
+        # the graph is traced in full precision and compressed afterwards.
+        precision = _OPENVINO_PRECISION if use_openvino else config.precision
```
Comment (mpryahin) [inline application/backend/app/runtime/core/components/factories/model.py:130]: Do we still need the `use_openvino` option? It was added as a feature flag when openvino support was experimental

File: application/backend/app/runtime/core/components/factories/model.py [hunk 3]
```diff
@@ -1,145 +1,455 @@
...
+                precision=config.precision,
+                device=device_info.as_torch,
+                prompt_mode=prompt_mode,
+            )
+
+        model.fit(reference_batch)
+        return model
+
+    @staticmethod
+    def _resolve_sam3_ir(config: Sam3Config, compression: CompressionMode) -> Path:
```
Comment (mpryahin) [inline application/backend/app/runtime/core/components/factories/model.py:280]: There are quite a few model export details exposed in the factory class. This model factory logic reflects all the steps a user needs to do when starting with a Torch model.

I was wondering whether we could encapsulate these details in the model’s export method instead, so that the client-side flow is as simple as creating a Torch model instance, exporting it, and then creating an OpenVINO model instance.

File: application/backend/app/runtime/core/components/factories/model.py [hunk 4]
```diff
@@ -1,145 +1,455 @@
...
+            model_id=SAM3_MODEL_ID,
+        )
+        try:
+            return exporter.to_openvino(export_root, ExportConfig(compression=compression))
+        finally:
+            del exporter
+            empty_accelerator_cache()
+
+
+def _compression_mode(config: ModelConfig) -> CompressionMode:
```
Comment (mpryahin) [inline application/backend/app/runtime/core/components/factories/model.py]: Could it be the config logic?

**Review by michalad1 — APPROVED** [review]
LGTM
